src/edu/stanford/dlss/wowza/SulWowza.java

```diff
@@ -15,73 +17,203 @@
+        else
+        {
+            getLogger().error(this.getClass().getSimpleName() + ": unable to parse druid and filename from streamName " + streamName);
+            return false;
+        }
+    }
+
+    static private final Pattern DRUID_REGEX_PATTERN = Pattern.compile("[a-z]{2}\\d{3}[a-z]{2}\\d{4}");
+
+    /** the druid of the media file to be streamed;  needed to verify stacks token
+     * Assumption:  validateStreamName() was already called */
+    String getDruid(String streamName)
+    {
+        String[] pieces = streamName.split("/");
+        String druid = pieces[0] + pieces[1] + pieces[2] + pieces[3];
+        if (DRUID_REGEX_PATTERN.matcher(druid).matches())
+            return druid;
+        else
+            return null;
+    }
+
+    /** the media file to be streamed, as stored under the druid-tree;  needed to verify stacks token
+     * Assumption:  validateStreamName() was already called */
+    String getFilename(String streamName)
+    {
+        String filename = streamName.substring(streamName.lastIndexOf('/') + 1);
+        if (filename.length() > 0)
+            return filename;
+        else
+            return null;
+    }
+
+    /** Assumption: stacksToken, druid, userIp and filename are all reasonable values (non-null, not empty, etc.) */
+    String getVerifyStacksTokenUrl(String stacksToken, String druid, String filename, String userIp)
+    {
+        // TODO:  Url encode anything?   utf-8 charset affect anything? (filename, stacksToken)
+        String queryStr = "stacks_token=" + stacksToken + "&user_ip=" + userIp;
+        String fullUrl = stacksTokenVerificationBaseUrl + "/media/" + druid + "/" + filename + "/verify_token?" + queryStr;
+        try
+        {
+            new URL(fullUrl);
+            return fullUrl;
+        }
+        catch (MalformedURLException err)
+        {
+            getLogger().error(this.getClass().getSimpleName() + " bad URL for stacks_token verification: ", err);
+            return null;
+        }
+    }
+
+    /** Assumption: verifyStacksTokenRequestUrl is a valid URL */
+    boolean verifyStacksToken(String verifyStacksTokenUrl)
+    {
+        try
+        {
+            URL tokenVerifyURL = new URL(verifyStacksTokenUrl);
+            URLConnection stacksConn = tokenVerifyURL.openConnection();
+            stacksConn.connect();
+            int status = ((HttpURLConnection) stacksConn).getResponseCode();
+            getLogger().info(this.getClass().getSimpleName() + " sent verify_token request to " + verifyStacksTokenUrl);
+            getLogger().debug(this.getClass().getSimpleName() + " verify_token response code is " + String.valueOf(status));
+            if (status == HttpURLConnection.HTTP_OK)
+                return true;
+            else
+                return false;
+        }
+        catch (IOException e)
+        {
+            getLogger().error(this.getClass().getSimpleName() + " unable to verify stacks token at " + verifyStacksTokenUrl);
```

Comment:
Can we print the error message from the exception itself here? Just like we do above on line 192? It doesn't hurt to output all of the available information while we're at it with these error messages.